internal/wait/wait.go

```diff
@@ -41,6 +42,34 @@ func ClusterPolicyReady(apiClient *clients.Settings, clusterPolicyName string, p
 		})
 }
 
+// NicClusterPolicyReady Waits until nicClusterPolicy is Ready.
+func NicClusterPolicyReady(apiClient *clients.Settings, nicClusterPolicyName string, pollInterval,
+	timeout time.Duration) error {
+	return wait.PollUntilContextTimeout(
+		context.TODO(), pollInterval, timeout, true, func(ctx context.Context) (bool, error) {
+			nicClusterPolicy, err := nvidianetwork.Pull(apiClient, nicClusterPolicyName)
+
+			if err != nil {
+				glog.V(gpuparams.GpuLogLevel).Infof("ClusterPolicy pull from cluster error: %s\n", err)
+
+				return false, err
+			}
+
+			if nicClusterPolicy.Object.Status.State == "ready" {
```

Comment:
You could use the constant defined in the API: 
```go
import (
      networkoperator "github.com/Mellanox/network-operator/api/v1alpha1"
)
```

and then `nicClusterPolicy.Object.Status.State == networkoperator.StateReady`

Reply:
BTW there's another useful constant that can be used throughout the code `networkoperator.NicClusterPolicyCRDName`, with the value of "NicClusterPolicy"

Reply:
Updated.  That's great input, thank you ! We need optimize the constants across the code, I agree, I will address in next PR when we optimize and create global variables common to both gpu and network operators may be create nvidiaci package.  Trying not to make updates to the GPU operator code now in this PR.